sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelation.scala

```diff
@@ -160,6 +160,9 @@ private[sql] case class InsertIntoHadoopFsRelation(
       logInfo("Skipping insertion into a relation that already exists.")
     }
 
+    // Invalidate the cache.
```

Comment:
@yhuai We need to refresh the `relation` to get the latest file status right here, however, we couldn't find the correct relation from `catalog`, probably we need to provide a API for `catalog`, which support get the relation by `data path`.

And even user refresh the file status explicitly, I don't think we have correct API for that, do we?